src/main/java/com/igot/cb/discussion/service/impl/DiscussionServiceImpl.java

```diff
@@ -1236,4 +1244,138 @@ private List<Map<String, Object>> fetchAndEnhanceDiscussions(List<Map<String, Ob
         }
         return filteredDiscussions;
     }
+
+    @Override
+    public ApiResponse searchDiscussionByCommunity(String communityId, Map<String, Object> paginationParams) {
+        log.info("DiscussionServiceImpl::searchDiscussionByCommunity");
+        ApiResponse response = ProjectUtil.createDefaultResponse("search.discussion.by.community");
+
+        if (StringUtils.isBlank(communityId)) {
+            createErrorResponse(response, Constants.INVALID_COMMUNITY_ID, HttpStatus.BAD_REQUEST, Constants.FAILED_CONST);
+            return response;
+        }
+
+        String errorMsg = validatePaginationParams(paginationParams);
+        if (!errorMsg.isEmpty()) {
+            createErrorResponse(response, errorMsg, HttpStatus.BAD_REQUEST, Constants.FAILED_CONST);
+            return response;
+        }
+
+        SearchCriteria searchCriteria = getCriteria((int) paginationParams.getOrDefault(Constants.PAGE_NUMBER, 0), (int) paginationParams.getOrDefault(Constants.PAGE_SIZE, 10));
+        String cacheKey = Constants.DISCUSSION_CACHE_PREFIX + communityId + "_" + searchCriteria.getPageNumber();
+        SearchResult searchResult = redisTemplate.opsForValue().get(cacheKey);
+        if (searchResult != null) {
+            log.info("DiscussionServiceImpl::searchDiscussionByCommunity: search result fetched from redis");
+            response.getResult().put(Constants.SEARCH_RESULTS, searchResult);
+            createSuccessResponse(response);
+            return response;
+        }
```

Comment:
cache is not possible for this - as we may require to exclude discussion post for certain user based on flagged by user.